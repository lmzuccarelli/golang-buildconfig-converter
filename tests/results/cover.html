
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>service: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go (93.3%)</option>
				
				<option value="file1">github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go (94.7%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package service

import (
        "fmt"
        "log"
        "os"
)

func Convert(config string) error <span class="cov8" title="1">{

        cfg, err := readConfigFile(config)
        if err != nil </span><span class="cov8" title="1">{
                return err
        }</span>
        <span class="cov8" title="1">log.Printf("deleting working directory %s", cfg.Spec.WorkingDirectory)
        os.RemoveAll(cfg.Spec.WorkingDirectory)
        log.Printf("creating working directory %s", cfg.Spec.WorkingDirectory)
        os.MkdirAll(cfg.Spec.WorkingDirectory+"/manifests", os.ModePerm)

        bcs, err := readAllBuildConfigs(cfg.Spec.BuildConfigPath)
        fmt.Println("DEBUG LMZ ", bcs)
        if err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">err = convertBuildConfigs(cfg, bcs)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package service

import (
        "fmt"
        "io/ioutil"

        "github.com/luigizuccarelli/golang-buildconfig-converter/pkg/schema"
        "gopkg.in/yaml.v2"
)

func mapToBuild(bc schema.BuildConfig) schema.Build <span class="cov8" title="1">{
        b := schema.Build{}
        b.Kind = "Build"
        b.APIVersion = bc.APIVersion
        b.Metadata.Name = bc.Metadata.Name
        return b
}</span>

func readAllBuildConfigs(dir string) ([]schema.BuildConfig, error) <span class="cov8" title="1">{
        var bcs []schema.BuildConfig
        var bc *schema.BuildConfig
        files, err := ioutil.ReadDir(dir)
        if err != nil </span><span class="cov8" title="1">{
                return []schema.BuildConfig{}, err
        }</span>
        <span class="cov8" title="1">for _, f := range files </span><span class="cov8" title="1">{
                if !f.IsDir() </span><span class="cov8" title="1">{
                        yfile, err := ioutil.ReadFile(dir + "/" + f.Name())
                        if err != nil </span><span class="cov0" title="0">{
                                return []schema.BuildConfig{}, err
                        }</span>
                        <span class="cov8" title="1">err = yaml.Unmarshal(yfile, &amp;bc)
                        if err != nil </span><span class="cov8" title="1">{
                                return []schema.BuildConfig{}, err
                        }</span>
                        // we are only interested in BuildConfigs
                        <span class="cov8" title="1">if bc.Kind == "BuildConfig" </span><span class="cov8" title="1">{
                                bcs = append(bcs, *bc)
                        }</span>
                }
        }
        <span class="cov8" title="1">return bcs, nil</span>
}

func readConfigFile(file string) (*schema.Config, error) <span class="cov8" title="1">{
        var cfg *schema.Config
        yfile, err := ioutil.ReadFile(file)
        if err != nil </span><span class="cov8" title="1">{
                return &amp;schema.Config{}, err
        }</span>
        <span class="cov8" title="1">err = yaml.Unmarshal(yfile, &amp;cfg)
        if err != nil </span><span class="cov8" title="1">{
                return &amp;schema.Config{}, err
        }</span>
        <span class="cov8" title="1">return cfg, nil</span>
}

func convertBuildConfigs(cfg *schema.Config, bcs []schema.BuildConfig) error <span class="cov8" title="1">{
        for _, bc := range bcs </span><span class="cov8" title="1">{
                b := mapToBuild(bc)
                yml, err := yaml.Marshal(b)
                if err != nil </span><span class="cov0" title="0">{
                        return fmt.Errorf("converting struct to yaml for %s", bc.Metadata.Name)
                }</span>
                <span class="cov8" title="1">err = ioutil.WriteFile(cfg.Spec.WorkingDirectory+"/manifests/"+bc.Metadata.Name+".yaml", yml, 0755)
                if err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
