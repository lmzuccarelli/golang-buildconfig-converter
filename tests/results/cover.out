mode: set
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:14.35,17.16 2 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:20.2,26.16 6 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:30.2,31.16 2 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:34.2,34.12 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:17.16,19.3 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:26.16,28.3 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/convert.go:31.16,33.3 1 0
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:20.53,31.2 10 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:33.68,37.16 4 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:40.2,40.26 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:56.2,56.17 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:37.16,39.3 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:40.26,41.17 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:41.17,43.18 2 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:46.4,47.18 2 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:51.4,51.30 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:43.18,45.5 1 0
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:47.18,49.5 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:51.30,53.5 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:59.58,62.16 3 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:65.2,66.16 2 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:69.2,69.17 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:62.16,64.3 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:66.16,68.3 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:72.78,73.25 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:84.2,84.12 1 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:73.25,76.17 3 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:79.3,80.17 2 1
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:76.17,78.4 1 0
github.com/luigizuccarelli/golang-buildconfig-converter/pkg/service/helper.go:80.17,82.4 1 1
